app/src/main/java/pm/gnosis/heimdall/ui/transactions/BaseTransactionActivity.kt

```diff
@@ -91,14 +91,16 @@ abstract class BaseTransactionActivity : BaseActivity() {
         }
     }
 
-    protected fun <T> checkInfoAndPerform(info: Pair<BigInteger?, Result<Transaction>>, action: (BigInteger, Transaction) -> Observable<Result<T>>): Observable<Result<T>> =
-            info.let { (safeAddress, transaction) ->
-                safeAddress?.let {
-                    when (transaction) {
-                        is DataResult -> action(safeAddress, transaction.data)
-                        is ErrorResult -> Observable.just(ErrorResult(transaction.error))
-                    }
-                } ?: Observable.just(ErrorResult(NoSafeSelectedException()))
+    protected fun <T> Observable<Pair<BigInteger?, Result<Transaction>>>.checkedFlatMap(action: (BigInteger, Transaction) -> Observable<Result<T>>): Observable<Result<T>> =
+            flatMap {
+                it.let { (safeAddress, transaction) ->
+                    safeAddress?.let {
+                        when (transaction) {
+                            is DataResult -> action(safeAddress, transaction.data)
+                            is ErrorResult -> Observable.just(ErrorResult(transaction.error))
```

Comment:
Why not use the same result (transaction)?

Reply:
it's a different generic, so you cannot use the same result without an unsafe case